app/services/update_projects_types_service.rb

```diff
@@ -0,0 +1,59 @@
+#-- encoding: UTF-8
+#-- copyright
+# OpenProject is a project management system.
+# Copyright (C) 2012-2015 the OpenProject Foundation (OPF)
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License version 3.
+#
+# OpenProject is a fork of ChiliProject, which is a fork of Redmine. The copyright follows:
+# Copyright (C) 2006-2013 Jean-Philippe Lang
+# Copyright (C) 2010-2013 the ChiliProject Team
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License
+# as published by the Free Software Foundation; either version 2
+# of the License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+#
+# See doc/COPYRIGHT.rdoc for more details.
+#++
+
+UpdateProjectsTypesService = Struct.new :project do
```

Comment:
It obviously breaks inheritance since it implicitly returns an instance. It also does not add weird things to the ancestor chain , it just extends Enumerable to access all members of the struct. 

``` ruby
x = Struct.new :foo
irb(main):012:0> x.ancestors
=> [#<Class:0x007ff3ec3cee38>, Struct, Enumerable, Object, Kernel, BasicObject]

class Foo; end
irb(main):013:0> Foo.ancestors
=> [Foo, Object, Kernel, BasicObject]
```

I agree with the required arguments case, though.